packages/yew/src/platform/fmt/buffer.rs

```diff
@@ -0,0 +1,212 @@
+        if self.buf.is_empty() {
+            self.buf.reserve(BUF_SIZE);
+        }
+    }
+}
+
+impl Write for Inner {
+    fn write_str(&mut self, s: &str) -> fmt::Result {
+        if s.is_empty() {
+            return Ok(());
+        }
+
+        self.wake();
+        if s.len() < BUF_SIZE {
+            self.buf_reserve();
+        }
+
+        self.buf.write_str(s)
+    }
+
+    fn write_char(&mut self, c: char) -> fmt::Result {
+        self.wake();
+        self.buf_reserve();
+
+        self.buf.write_char(c)
+    }
+
+    fn write_fmt(&mut self, args: fmt::Arguments<'_>) -> fmt::Result {
+        self.wake();
+        self.buf_reserve();
+
+        self.buf.write_fmt(args)
+    }
+}
+
+/// An asynchronous String BufWriter.
+///
+/// This type implements [`fmt::Write`] and can be.
+pub(crate) struct BufWriter {
+    inner: Rc<UnsafeCell<Inner>>,
+}
+
+impl Write for BufWriter {
+    #[inline]
+    fn write_str(&mut self, s: &str) -> fmt::Result {
+        // SAFETY:
+        //
+        // We can acquire a mutable reference without checking as:
+        //
+        // - This type is !Sync and !Send.
+        // - This function is not used by any other functions that has access to the inner type.
+        // - The mutable reference is dropped at the end of this function.
+        let inner = unsafe { &mut *self.inner.get() };
+
+        inner.write_str(s)
+    }
+
+    #[inline]
+    fn write_char(&mut self, c: char) -> fmt::Result {
+        // SAFETY:
+        //
+        // We can acquire a mutable reference without checking as:
+        //
+        // - This type is !Sync and !Send.
+        // - This function is not used by any other functions that has access to the inner type.
+        // - The mutable reference is dropped at the end of this function.
+        let inner = unsafe { &mut *self.inner.get() };
+
+        inner.write_char(c)
+    }
+
+    #[inline]
+    fn write_fmt(&mut self, args: fmt::Arguments<'_>) -> fmt::Result {
+        // SAFETY:
+        //
+        // We can acquire a mutable reference without checking as:
+        //
+        // - This type is !Sync and !Send.
+        // - This function is not used by any other functions that has access to the inner type.
+        // - The mutable reference is dropped at the end of this function.
+        let inner = unsafe { &mut *self.inner.get() };
+
+        inner.write_fmt(args)
+    }
+}
+
+impl Drop for BufWriter {
+    fn drop(&mut self) {
+        // SAFETY:
+        //
+        // We can acquire a mutable reference without checking as:
+        //
+        // - This type is !Sync and !Send.
+        // - This function is not used by any other functions that has access to the inner type.
+        // - The mutable reference is dropped at the end of this function.
+        let inner = unsafe { &mut *self.inner.get() };
+
+        inner.wake();
+        inner.state = BufStreamState::Done;
+    }
+}
+
+/// An asynchronous String Stream.
```

Comment:
Nit: Just some links in the docs. It doesn't really matter right now though since this is a private type but will when yew-platform becomes independent #2839 
```suggestion
/// An asynchronous [`String`] [`Stream`].
```